Over Head Icon improvements

TimGoll opened the pull request:
This PR brings multiple small and minor improvements to the over head icons:

- they track the head bone positions: This means they move with the player animation, move down when crouched and also change position when the model scale is changed (i.e. when the minifier is used)
- it now uses a proper Cam3D2D environment: This fixed some minor issues with the icon drop shadow
- rendering order is based on distance: Previously the icons were drawn in the player order which could cause icons to be rendered in front of icons that they shouldn't be; it is now sorted by distance
- when observing a player in their eyes, their icon is now hidden, it is still shown when in third person
- changed the angle the icon is drawn: the rotation is still the same, but the pitch is locked on 90 meaning that the arrow is always pointing at the head, even when looking from the top

![image](https://github.com/TTT-2/TTT2/assets/13639408/457d4594-39a9-4748-a17b-7899d180358e)
Only a trained eye can spot a visual difference here

saibotk commented on gamemodes/terrortown/gamemode/client/cl_targetid.lua:
```diff
@@ -53,54 +52,37 @@ local materialBaseOverlay = Material("vgui/ttt/dynamic/sprite_base_overlay")
-		return
-	end
+function DrawOverheadRoleIcon(client, ply, iconRole, colorRole)
+	local pos = ply:GetBoneMatrix(ply:LookupBone("ValveBiped.Bip01_Head1")):GetTranslation()
```
What if a model does not have this? Can we just add a fallback position then?

TimGoll replied:
good idea, I will add the old code as a fallback here

saibotk commented on gamemodes/terrortown/gamemode/client/cl_targetid.lua:
```diff
@@ -53,54 +52,37 @@ local materialBaseOverlay = Material("vgui/ttt/dynamic/sprite_base_overlay")
-	-- stop linear filter
-	render.PopFilterMag()
-	render.PopFilterMin()
+	return client:GetPos():Distance(a.ply:GetPos()) > client:GetPos():Distance(b.ply:GetPos())
```
Cache client:GetPos instead of client here

saibotk reviewed (changes requested):
Just minor nits

Histalek commented on CHANGELOG.md:
```diff
@@ -244,6 +244,10 @@ All notable changes to TTT2 will be documented here. Inspired by [keep a changel
+- Changed the way the role overhead icon is rendered (by @TimGoll)
+  - It now tracks the players head position
+  - Rendering order is based on distance, no more weird visual glitches
+  - Hidden when oberserving a player in their eyes
```
```suggestion
  - Hidden when observing a player in first person view
```

TimGoll commented:
Done @saibotk! While adding the fallback, I moved the code to the player class so that we now have a function that tells the exact height of the player. It is not perfect because I assumed a head size, but it is better then it was before:

```lua
---
-- Returns the player height vector based on the curren thead bone
-- position. X and Y move along the head bone, Z contains the actual
-- height.
-- @note Uses the player's bounding box as a fallback if the head
-- bone is not defined
-- @note Respects the model scale for the height calculation
-- @return vector The player height
-- @realm shared
function plymeta:GetHeightVector()
```

I also noticed an issue in the draw check code that I fixed: 
![image](https://github.com/TTT-2/TTT2/assets/13639408/d71689d4-723a-4e3a-910a-2d9b43195b7e)

TimGoll commented:
In the last commit I added support for the head bone scale used in Alf's minigames:

![image](https://github.com/TTT-2/TTT2/assets/13639408/0cb0a4cd-7e07-42b8-b387-af6a43b3b2ea)
